### PR TITLE
Issue #764: Limit min date for one-time pickups to today

### PR DESCRIPTION
**Pull request by timnovis**
Fixes issue #764

### CODE REVIEW

File: src/components/Pickups/PickupEdit.vue [hunk 1]
```diff
@@ -21,7 +21,7 @@
         :error="hasError('date')"
         :error-label="firstError('date')"
         >
-        <q-datetime type="date" v-model="edit.date" :display-value="$d(edit.date, 'dateShort')"/>
+        <q-datetime type="date" v-model="edit.date" :min="this.today" :display-value="$d(edit.date, 'dateShort')"/>
```
Comment (nicksellen) [inline src/components/Pickups/PickupEdit.vue]: You can leave out the `this.` in templates

File: src/mixins/editMixin.js [hunk 2]
```diff
@@ -26,14 +26,22 @@ export default {
     hasChanged () {
       return !this.isNew && !deepEqual(this.value, this.edit)
     },
+    today () {
```
Comment (nicksellen) [inline src/mixins/editMixin.js]: Builtin classes (e.g. `Date`) are not reactive, so this will never update.

You could either just initialize it in the `data()` function (it doesn't really _need_ to be reactive).

Or, if you want a reactive value, you can get it using:

```js
import dateFnsHelper from '@/services/dateFnsHelper'

// ... stuff

computed: {
  now () {
    return dateFnsHelper.now // <--- this value is reactive!
  }
}
```

File: src/mixins/editMixin.js [hunk 3]
```diff
@@ -26,14 +26,22 @@ export default {
...
       if (this.isNew) {
         this.$emit('save', this.edit, event)
       }
       else {
-        this.$emit('save', { ...objectDiff(this.value, this.edit), id: this.value.id }, event)
+        this.$emit(
+          'save',
+          { ...objectDiff(this.value, this.edit), id: this.value.id },
+          event,
+        )
```
Comment (nicksellen) [inline src/mixins/editMixin.js]: Could you revert this unrelated formatting change?

File: src/components/Pickups/PickupEdit.vue (hunk 1 above)
Comment (timnovis) [reply src/components/Pickups/PickupEdit.vue]: Oops! I knew that. Sorry.

File: src/mixins/editMixin.js (hunk 2 above)
Comment (timnovis) [reply src/mixins/editMixin.js]: Fix pushed

File: src/mixins/editMixin.js (hunk 3 above)
Comment (timnovis) [reply src/mixins/editMixin.js]: Fix pushed

File: src/components/Pickups/PickupEdit.vue (hunk 1 above)
Comment (timnovis) [reply src/components/Pickups/PickupEdit.vue]: Fix pushed

File: src/mixins/editMixin.js (hunk 2 above)
Comment (nicksellen) [reply src/mixins/editMixin.js]: Ah, I just noticed this is in the `editMixin`. It would be more appropriate in the `PickupEdit` component as the past date only relates to that, not editing in general.

Also, when you move it, maybe rename the computed property to `now` so it's consistent with the name inside the helper. Thanks!

**Comment by tiltec** [comment]
Thanks for your initiative! However, did you notice that @D0nPiano already assigned himself to the issue and proposed a fix as part of #769?

**Comment by nicksellen** [comment]
Yeah, but this one is :sparkles: _responsive_ :sparkles:

**Comment by timnovis** [comment]
Updated!

**Comment by tiltec** [comment]
True, I'd always favor a reactive solution :)
